lib/models/LifecycleConfiguration.ts

```diff
@@ -852,13 +885,18 @@ export default class LifecycleConfiguration {
             }
 
             actions.forEach(action => {
-                const { actionName, days, date, deleteMarker } = action;
+                const { actionName, days, date, deleteMarker, newerNoncurrentVersions } = action;
                 if (actionName === 'AbortIncompleteMultipartUpload') {
                     entry.addAbortMPU(days!);
                     return;
                 }
                 if (actionName === 'NoncurrentVersionExpiration') {
-                    entry.addNCVExpiration(days!);
+                    entry.addNCVExpiration('NoncurrentDays', days!);
+
+                    if (newerNoncurrentVersions) {
+                        entry.addNCVExpiration('NewerNoncurrentVersions', newerNoncurrentVersions);
+                    }
```

Comment:
If the above comment makes sense:

```suggestion
                    entry.addNCVExpiration('NewerNoncurrentVersions', newerNoncurrentVersions!);
```